jujugui/static/gui/src/app/components/entity-details/entity-details.js

```diff
@@ -18,14 +18,111 @@ with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 'use strict';
 
-YUI.add('entity-details', function() {
+YUI.add('entity-details', function(Y) {
+
+  var utils = Y.namespace('juju.views.utils');
 
   juju.components.EntityDetails = React.createClass({
+    fetchSuccess: function(models) {
+      this.setState({waitingForFetch: false});
+      if (models.length > 0) {
+        this.setState({entity: models[0].toEntity()});
+      }
+    },
+
+    fetchFailure: function() {
+      // XXX Implement error handling.
+      console.error('Fetching the entity failed.');
+    },
+
+    fetchEntity: function(id) {
+      this.setState({waitingForFetch: true});
+      this.props.getEntity(
+        id,
+        this.fetchSuccess,
+        this.fetchFailure
+      );
+    },
+
+    shouldRefetch: function(nextProps) {
+      var entity = this.state.entity;
+      return !entity || nextProps.id !== entity.id;
+    },
+
+    getInitialState: function() {
+      return {
+        entity: null,
+        waitingForFetch: false
+      };
+    },
+
+    componentDidMount: function() {
+      this.fetchEntity(this.props.id);
+    },
+
+    componentWillReceiveProps: function(nextProps) {
+      if (this.shouldRefetch(nextProps)) {
+        this.fetchEntity(nextProps.id);
+      }
+    },
+
+    shouldComponentUpdate: function(nextProps, nextState) {
+      return this.shouldRefetch(nextProps) && !this.state.waitingForFetch;
+    },
+
     render: function() {
-      return <h1>Entity details</h1>;
+      var entity = this.state.entity,
+          tags = '';
+      if (!entity) {
+        return (
+          <div className="spinner-loader">Loading...</div>
+        );
+      }
+      if (entity.tags) {
+        tags = entity.tags.map(function(tag) {
+          return(
+            <li className="tag-list--item">
+              <a target="_blank" href={'/q?tags=' + tag}>{tag}</a>
+            </li>
+          );
+        });
+      }
+      return (
+        <div className={'entity-details ' + entity.type}>
+          <div className="row-hero">
+            <header className="twelve-col">
+              <div className="inner-wrapper">
+                <div className="eight-col no-margin-bottom">
+                  <img src={entity.iconPath} alt="" width="96" id="the-icon"/>
+                  <div className="header__details">
+                    <h1 className="header__title" itemprop="name">{entity.displayName}</h1>
+                    <p className="header__by">By <a href={'https://launchpad.net/~' + entity.owner} target="_blank">{entity.owner}</a></p>
+                    <ul className="bullets inline">
+                      <li className="revisions-item">
+                        <a href="#revisions" className="revisions-link">{entity.revisions.length} {utils.pluralize('revision', entity.revisions.length)}</a>
```

Comment:
This will also need to update state when `onClick` so right now we could just leave off the hash support.